src/quad-tree.ts

```diff
@@ -148,33 +148,34 @@ export class QuadTree<T extends Rectangle> {
    * Since this function performs an operation on the tree, it will also notify all subscribers that a delete operation has occured
    * and will provide the value passed to this function.
    */
-  public delete(at: Rectangle | Point) {
+  public delete(at: Rectangle | Point, shouldEmitEvent: boolean = true) {
```

Comment:
You are still passing a rectangle, so this doesn't accomplish anything other than no longer creating an extra function on the rectangle when it is made. This seems like a premature optimization to me. In addition, every change should trigger events, it is up to subscribers to determine whether the event is relevant. Think of it like a proxy, the caller of this function should not be aware that there are any subscribers in my opinion - the same as any other store that has a subscriber. If you want to avoid an infinite loop of events.